pathfinderlib/main.py

```diff
@@ -25,33 +63,212 @@ def make_fastq_to_fasta_command(self, filename):
             ]
 
     def fastq_to_fasta(self, filename):
-        fasta = tempfile.NamedTemporaryFile(delete=False)
-        command = self.make_fastq_to_fasta_command(filename)
+        fasta = tempfile.NamedTemporaryFile()
+        command = self._make_fastq_to_fasta_command(filename)
         subprocess.check_call(command, stdout=fasta, stderr=subprocess.STDOUT)
-        return fasta.name
+        return fasta
 
-    def make_blastx_command(self, R, rex, output):
-        out_dir = os.path.join(self.config["humann_fp"], "input", output + rex)
-        return [
-            "blastx", "-outfmt", "6",
-            "-db", self.config["kegg_fp"],
-            "-query", R, "-out", out_dir
-            ]
+    def make_command(self, R, output):
+        raise NotImplementedError("Create and override method in child tool")
 
-    def run_blastx(self, R, rex, output):
+    def run(self, R, out_dir):
         r_fasta = self.fastq_to_fasta(R)
-        command = self.make_blastx_command(r_fasta, rex, out_dir)
+        output_fp = self.make_output_fp(out_dir, R)
+        command = self.make_command(r_fasta.name, output_fp)
         subprocess.check_call(command, stderr=subprocess.STDOUT)
-        os.remove(r_fasta)
+        return output_fp+'.m8' if self.search_method.lower()=='rapsearch' else output_fp
```

Comment:
I wonder if we could work this out to remove the if statement.  Maybe add a method `get_alignment_fp()` that returns `output_fp` for BLASTX but `output_fp + ".m8"` for rapsearch?